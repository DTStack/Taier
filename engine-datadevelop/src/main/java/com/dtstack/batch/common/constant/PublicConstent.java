package com.dtstack.batch.common.constant;

/**
 * @company: www.dtstack.com
 * @Author ：Nanqi
 * @Date ：Created in 12:38 2020/3/2
 * @Description：公有常量
 */
public interface PublicConstent {
    /**
     * UPSERT 魔法字符
     */
    public static String UPSERT = "upsert";

    public static String AUTH_EH_CACHE = "BatchAuthSpringCache";
}
