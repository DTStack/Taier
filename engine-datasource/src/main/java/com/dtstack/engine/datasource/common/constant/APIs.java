package com.dtstack.engine.datasource.common.constant;

/**
 * Created by 袋鼠云-数栈产研部-应用研发中心.
 *
 * @author <a href="mailto:linfeng@dtstack.com">林丰</a>
 * @date 2021/3/4
 * @desc API常量
 */
public interface APIs {

    String PREFIX = "/api/publicService";
    String OPEN_PREFIX = PREFIX + "/open";
}
