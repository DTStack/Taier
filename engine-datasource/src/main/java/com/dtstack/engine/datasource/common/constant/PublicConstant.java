package com.dtstack.engine.datasource.common.constant;

/**
 * 公共常量定义
 * @description:
 * @author: liuxx
 * @date: 2021/3/22
 */
public interface PublicConstant {
    String AUTH_EH_CACHE = "PublicSVCAuthSpringCache";

    Integer NO = 0;

    Integer YES = 1;
}
