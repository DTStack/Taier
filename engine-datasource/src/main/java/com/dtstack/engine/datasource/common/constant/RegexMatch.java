package com.dtstack.engine.datasource.common.constant;

/**
 * 正则表达式常量类
 * @description:
 * @author: liuxx
 * @date: 2021/3/17
 */
public interface RegexMatch {

    /**
     * HDFS正则表达式
     */
    String DEFAULT_FS_REGEX = "hdfs://.*";

}
