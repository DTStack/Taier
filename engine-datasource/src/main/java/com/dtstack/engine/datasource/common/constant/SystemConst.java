package com.dtstack.engine.datasource.common.constant;

/**
 * Created by 袋鼠云-数栈产研部-应用研发中心.
 *
 * @author <a href="mailto:linfeng@dtstack.com">林丰</a>
 * @date 2021/3/3
 * @desc 常量类示例
 */
public class SystemConst {

    public static final String LEFT_SLASH = "/";
    public static final String DOUBLE_SLASH = "//";
    /**
     * 产品授权
     */
    public static final Integer IS_PRODUCT_AUTH = 1;
    /**
     * 产品取消授权
     */
    public static final Integer NOT_IS_PRODUCT_AUTH = 0;
}
