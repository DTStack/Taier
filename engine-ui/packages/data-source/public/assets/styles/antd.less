@import '~antd/dist/antd.less';

@primary-color: #3f87ff; // 全局主色
@success-color: #12bc6a; // 成功色
@warning-color: #ffb310; // 警告色
@error-color: #ff5f5c; // 错误色
@font-size-base: 12px; // 主字号
@link-color: #5c99ff; // 链接色
@heading-color: #333; // 标题色
@text-color: #333; // 主文本色
@text-color-secondary: #666; // 次文本色
@disabled-color: #bfbfbf; // 失效色
@border-color-base: #ddd; // 边框色
// @box-shadow-base: 0 2px 8px rgba(0, 0, 0, 0.15); // 浮层阴影

// antd 分页器样式
.ant-pagination-item-active {
	a {
		font-family: PingFangSC-Regular;
		letter-spacing: 0;
		text-align: center;
	}
}

.ant-pagination {
	position: relative;
	left: 9px;
	box-sizing: border-box;
	margin: 0;
	margin: 14px 10px 0 0;
	padding: 0;
	color: rgba(0, 0, 0, 0.85);
	font-size: 14px;
	font-variant: tabular-nums;
	line-height: 1.5715;
	list-style: none;
	font-feature-settings: 'tnum';
}

.ant-table-cell {
	border-right: none !important;
}
.ant-table-measure-row td > div {
	display: none !important;
}
.dt-notification-close-btn {
	position: absolute !important;
	top: 18px !important;
	right: 46px;
	font-weight: normal;
	font-size: 13px;
	line-height: 17px;
	user-select: none;
	&:hover {
		color: rgb(39, 39, 39);
	}
}
.anticon-eye-invisible .ant-input-password-icon {
	color: #999;
}
.anticon-close .ant-select-remove-icon {
	color: #999;
}

// 输入框，下拉框禁用样式
.ant-input[disabled] {
	color: #999999 !important;
	background-color: #e8e8e8 !important;
	border: 1px solid #dddddd;
}

.ant-select-disabled {
	&:hover {
		border: none;
	}
	.ant-select-selection {
		color: #999999 !important;
		background-color: #e8e8e8 !important;
		border: 1px solid #dddddd;
	}
}

.ant-form-item {
	margin-bottom: 20px;

	&.ant-form-item-with-help {
		margin-bottom: 4px;
	}

	label {
		top: 50%;
		display: inline-block;
		transform: translateY(-50%);
	}

	.ant-form-item-label {
		height: 32px;
	}

	.ant-form-item-label,
	.ant-form-item-control {
		line-height: unset;
	}

	textarea.ant-input {
		margin-bottom: 0;
	}
}

.ant-modal {
	height: 80vh;

	.ant-modal-content {
		max-height: 80vh;
	}

	.ant-modal-body {
		max-height: calc(80vh - 127px);
		padding-bottom: 28px;
		overflow-y: auto;
	}

	.ant-modal-footer,
	.ant-modal-body,
	.ant-modal-header {
		padding-right: 20px;
		padding-left: 20px;
		border-color: #ddd;
	}

	.ant-modal-footer {
		padding: 19.5px 16px;
	}
}

.ant-table-tbody > tr > td {
	vertical-align: middle;
}

.ant-btn:hover,
.ant-btn:focus,
.ant-btn:active,
.ant-btn.active {
	color: #333;
	border-color: #bfbfbf;
}
.ant-btn.ant-btn-primary:hover,
.ant-btn.ant-btn-primary:focus {
	color: #fff;
	background-color: #5c99ff;
	border-color: #5c99ff;
}
.ant-btn.ant-btn-background-ghost:hover,
.ant-btn.ant-btn-background-ghost:focus {
	color: #fff;
	background-color: #5c99ff !important;
	border-color: #5c99ff;
}

.ant-message-notice-content {
	height: auto !important;
	.ant-message-custom-content {
		height: auto !important;
	}
}
