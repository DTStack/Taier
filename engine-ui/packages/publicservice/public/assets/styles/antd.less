@import '~antd/dist/antd.less';
@main-color: #268BFC;

// antd 分页器样式
.ant-pagination-item {
  &:hover {
    border-color: @main-color  !important;

    a {
      color: @main-color  !important;
    }
  }
}

.ant-pagination-item-active {
  background: @main-color  !important;

  a {
    font-family: PingFangSC-Regular;
    font-size: 14px;
    color: #ffffff !important;
    letter-spacing: 0;
    text-align: center;
  }

  &:hover {
    border-color: @main-color  !important;

    a {
      color: #ffffff !important;
    }
  }
}

.ant-pagination {
  box-sizing: border-box;
  margin: 0;
  padding: 0;
  position: relative;
  left: 9px;
  color: rgba(0, 0, 0, 0.85);
  font-size: 14px;
  font-variant: tabular-nums;
  line-height: 1.5715;
  list-style: none;
  font-feature-settings: 'tnum';
  // margin: 10px;
  margin: 14px 10px 0 0;
}

.ant-table-cell {

  border-right: none !important;

}
.ant-table-measure-row td>div{
	display: none !important;
}