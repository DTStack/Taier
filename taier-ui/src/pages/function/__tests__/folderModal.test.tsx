import { CATALOGUE_TYPE } from '@/constant';
import { cleanup, fireEvent, render, waitFor } from '@testing-library/react';
import { Input } from 'antd';
import FolderModal from '../folderModal';

jest.mock('../../../components/folderPicker', () => ({ id, value, onChange }: any) => (
    <Input id={id} value={value} onChange={onChange} data-testid="mockFolderPicker" />
));

describe('Test FolderModal Component for function', () => {
    beforeEach(() => {
        cleanup();
    });

    it('Match snapshot', () => {
        expect(
            render(
                <FolderModal
                    isModalShow
                    defaultData={undefined}
                    treeData={{} as any}
                    dataType={CATALOGUE_TYPE.FUNCTION}
                    toggleCreateFolder={jest.fn()}
                    addOfflineCatalogue={jest.fn()}
                    editOfflineCatalogue={jest.fn()}
                />
            ).asFragment()
        ).toMatchSnapshot();
    });

    it('Create folder from root folder', async () => {
        const fn = jest.fn(() => Promise.resolve(true));
        const { container } = render(
            <FolderModal
                isModalShow
                defaultData={undefined}
                treeData={{ id: 1 } as any}
                dataType={CATALOGUE_TYPE.FUNCTION}
                toggleCreateFolder={jest.fn()}
                addOfflineCatalogue={fn}
                editOfflineCatalogue={jest.fn()}
            />
        );

        const buttons = container.querySelectorAll<HTMLDivElement>('.ant-btn') || [];

        fireEvent.change(container.querySelector('input#dt_nodeName')!, {
            target: { value: 'test' },
        });

        fireEvent.click(buttons[1]);

        await waitFor(() => {
            expect(fn).toBeCalledWith({ nodeName: 'test', nodePid: 1 });
        });
    });

    it('Create folder from a specific folder', async () => {
        const fn = jest.fn(() => Promise.resolve(true));
        const { container } = render(
            <FolderModal
                isModalShow
                defaultData={{
                    parentId: 10,
                }}
                treeData={
                    {
                        id: 1,
                        children: [
                            {
                                id: 10,
                            },
                        ],
                    } as any
                }
                dataType={CATALOGUE_TYPE.FUNCTION}
                toggleCreateFolder={jest.fn()}
                addOfflineCatalogue={fn}
                editOfflineCatalogue={jest.fn()}
            />
        );

        const buttons = container.querySelectorAll<HTMLDivElement>('.ant-btn') || [];

        fireEvent.change(container.querySelector('input#dt_nodeName')!, {
            target: { value: 'test' },
        });

        fireEvent.click(buttons[1]);

        await waitFor(() => {
            expect(fn).toBeCalledWith({ nodeName: 'test', nodePid: 10 });
        });
    });

    it('Edit folder', async () => {
        const fn = jest.fn(() => Promise.resolve(true));
        const { container } = render(
            <FolderModal
                isModalShow
                defaultData={{
                    id: 10,
                    name: 'test',
                    parentId: 11,
                }}
                treeData={{} as any}
                dataType={CATALOGUE_TYPE.FUNCTION}
                toggleCreateFolder={jest.fn()}
                addOfflineCatalogue={jest.fn()}
                editOfflineCatalogue={fn}
            />
        );

        const buttons = container.querySelectorAll<HTMLDivElement>('.ant-btn') || [];

        fireEvent.change(container.querySelector('input#dt_nodeName')!, {
            target: { value: 'test2' },
        });

        fireEvent.click(buttons[1]);

        await waitFor(() => {
            expect(fn).toBeCalledWith({ id: 10, nodeName: 'test2', nodePid: 11, type: 'folder' });
        });
    });

    it('Close modal via button', () => {
        const fn = jest.fn();
        const { container } = render(
            <FolderModal
                isModalShow
                defaultData={undefined}
                treeData={{} as any}
                dataType={CATALOGUE_TYPE.FUNCTION}
                toggleCreateFolder={fn}
                addOfflineCatalogue={jest.fn()}
                editOfflineCatalogue={jest.fn()}
            />
        );
        const buttons = container.querySelectorAll<HTMLDivElement>('.ant-btn') || [];
        fireEvent.click(buttons[0]);

        expect(fn).toBeCalled();
    });
});
