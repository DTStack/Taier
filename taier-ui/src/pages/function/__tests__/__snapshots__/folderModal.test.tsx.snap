// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test FolderModal Component for function Match snapshot 1`] = `
<DocumentFragment>
  <div>
    <p
      data-testid="antd-mock-Modal-title"
    >
      新建文件夹
    </p>
    <form
      class="ant-form ant-form-horizontal"
    >
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
          >
            <label
              class="ant-form-item-required"
              for="dt_nodeName"
              title="目录名称"
            >
              目录名称
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  id="dt_nodeName"
                  placeholder="文件夹名称"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="ant-form-item"
      >
        <div
          class="ant-row ant-form-item-row"
        >
          <div
            class="ant-col ant-form-item-label ant-col-xs-24 ant-col-sm-6"
          >
            <label
              class=""
              title="选择目录位置"
            >
              选择目录位置
            </label>
          </div>
          <div
            class="ant-col ant-form-item-control ant-col-xs-24 ant-col-sm-16"
          >
            <div
              class="ant-form-item-control-input"
            >
              <div
                class="ant-form-item-control-input-content"
              >
                <input
                  aria-required="true"
                  class="ant-input"
                  id="nodePid"
                  type="hidden"
                  value=""
                />
                <input
                  class="ant-input"
                  data-testid="mockFolderPicker"
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <button
      class="ant-btn ant-btn-default ant-btn-lg"
      type="button"
    >
      <span>
        取 消
      </span>
    </button>
    <button
      class="ant-btn ant-btn-primary ant-btn-lg"
      type="button"
    >
      <span>
        确 认
      </span>
    </button>
  </div>
</DocumentFragment>
`;
